spring-boot/admin/src/main/java/eu/coatrack/admin/security/GatewayAuthFilter.java

```diff
@@ -44,24 +45,19 @@
  *
  * @author gr-hovest
  */
-public class GatewayAuthFilter extends GenericFilterBean {
+public class GatewayAuthFilter extends OncePerRequestFilter {
 
     private static final Logger log = LoggerFactory.getLogger(GatewayAuthFilter.class);
 
     @Override
-    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain) throws IOException, ServletException {
+    protected void doFilterInternal(HttpServletRequest servletRequest, HttpServletResponse servletResponse, FilterChain filterChain) throws ServletException, IOException {
 
-        if (servletRequest instanceof HttpServletRequest) {
-            final HttpServletRequest req = (HttpServletRequest) servletRequest;
-            final String keyParam = req.getParameter(Proxy.GATEWAY_API_KEY_REQUEST_PARAMETER_NAME);
+            final String keyParam = servletRequest.getParameter(Proxy.GATEWAY_API_KEY_REQUEST_PARAMETER_NAME);
```

Comment:
I think the name ```gatewayApiKeyParam``` would be more precisely.

Reply:
This is already outdated so done